python/lsst/ts/externalscripts/auxtel/latiss_cwfs_align.py

```diff
@@ -0,0 +1,547 @@
+
+        # angle between elevation axis and nasmyth2 rotator
+        self.angle = None
+
+        self.intra_visit_id = None
+        self.extra_visit_id = None
+
+        self.intra_exposure = None
+        self.extra_exposure = None
+        self.detection_exp = None
+
+        self.source_selection_result = None
+        self.cwfs_selected_sources = []
+
+        self.I1 = []
+        self.I2 = []
+        self.fieldXY = [0.0, 0.0]
+
+        self.inst = None
+        self.algo = None
+
+        self.zern = None
+        self.hexapod_corr = None
+
+        self.data_pool_sleep = 5.
+
+    @property
+    def dz(self):
+        if self._dz is None:
+            self.dz = 0.8
+        return self._dz
+
+    @property
+    def side(self):
+        return int(self._side * self.dz / 1.5)
+
+    @dz.setter
+    def dz(self, value):
+        self._dz = float(value)
+        cwfs_config_template = """#Auxiliary Telescope parameters:
+Obscuration 				0.3525
+Focal_length (m)			21.6
+Aperture_diameter (m)   		1.2
+Offset (m)				{}
+Pixel_size (m)				10.0e-6
+"""
+        config_index = f"auxtel_latiss"
+        path = Path(cwfs.__file__).resolve().parents[3].joinpath("data", config_index)
+        if not path.exists():
+            os.makedirs(path)
+        dest = path.joinpath(f"{config_index}.param")
+        with open(dest, "w") as fp:
+            fp.write(cwfs_config_template.format(self._dz * 0.041))
+        self.inst = Instrument(config_index, self.side * 2)
+        self.algo = Algorithm('exp', self.inst, 1)
+
+    async def take_intra_extra(self):
+        """ Take Intra/Extra images.
+
+        Returns
+        -------
+        images_end_readout_evt: list
+            List of endReadout event for the intra and extra images.
+
+        """
+
+        self.log.debug('Move to intra-focal position')
+
+        await self.hexapod_offset(-self.dz)
+
+        group_id = astropytime.Time.now().tai.isot
+
+        intra_image = await self.latiss.take_engtest(exptime=self.exposure_time, n=1,
+                                                     group_id=group_id,
+                                                     filter=self.filter, grating=self.grating)
+
+        self.log.debug('Move to extra-focal position')
+
+        await self.hexapod_offset(self.dz * 2.)
+
+        self.log.debug('Take extra-focal image')
+
+        extra_image = await self.latiss.take_engtest(exptime=self.exposure_time, n=1,
+                                                     group_id=group_id,
+                                                     filter=self.filter, grating=self.grating)
+
+        azel = await self.attcs.atmcs.tel_mount_AzEl_Encoders.aget()
+        nasmyth = await self.attcs.atmcs.tel_mount_Nasmyth_Encoders.aget()
+
+        self.angle = np.mean(azel.elevationCalculatedAngle) + np.mean(nasmyth.nasmyth2CalculatedAngle)
+
+        self.log.debug("Move hexapod to zero position")
+
+        await self.hexapod_offset(-self.dz)
```

Comment:
add a comment to say this puts the hexapod back to it's original (in-focus) position